client/knowledge_base_search/src/utils/replaceNewlines.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+export default function replaceNewlines(text) {
+  if (text.indexOf('\n') > 0){
+    text = text.split('\n').map((item, key, array) => {
+      return (
+                <span key={'newline_' + key}>
```

Comment:
Do you have any instances where you might be re-rendering these spans with new content within the same mounted component? If the new content shares the same key, it could cause it to not re-render properly.

Reply:
not in the same mounted component